packages/api/src/modules/providers/demarchesSimplifiees/demarchesSimplifiees.service.test.ts

```diff
@@ -575,4 +597,184 @@ describe("DemarchesSimplifieesService", () => {
+
+        beforeAll(() => {
+            sendQuerySpy = jest.spyOn(demarchesSimplifieesService, "sendQuery").mockResolvedValue(QUERY_RESULT);
+            generateSchemaInstructionSpy = jest
+                // @ts-expect-error -- spy private method
+                .spyOn(demarchesSimplifieesService, "generateSchemaInstruction")
+                // @ts-expect-error -- mock private method
+                .mockImplementation((c: DemarchesSimplifieesSingleSchemaSeed) => Promise.resolve({ from: c.to }));
+        });
+        jest.mocked(DemarchesSimplifieesDtoAdapter.toEntities).mockReturnValue([EXAMPLE]);
+
+        afterAll(() => {
+            sendQuerySpy.mockRestore();
+            generateSchemaInstructionSpy.mockRestore();
+            jest.mocked(DemarchesSimplifieesDtoAdapter.toEntities).mockRestore();
+        });
+
+        it("sends query", async () => {
+            await demarchesSimplifieesService.buildSchema(SCHEMA_SEED, FORM_ID);
+            expect(sendQuerySpy).toHaveBeenCalledWith(GetDossiersByDemarcheId, { demarcheNumber: FORM_ID });
+        });
+        it("adapts entities (to get example)", async () => {
+            await demarchesSimplifieesService.buildSchema(SCHEMA_SEED, FORM_ID);
+            expect(DemarchesSimplifieesDtoAdapter.toEntities).toHaveBeenCalledWith(QUERY_RESULT, FORM_ID);
+        });
+        it("finds id fo each field", async () => {
+            await demarchesSimplifieesService.buildSchema(SCHEMA_SEED, FORM_ID);
+            expect(generateSchemaInstructionSpy).toHaveBeenCalledTimes(2);
+        });
+        it("should returns aggregated results by field", async () => {
+            const actual = await demarchesSimplifieesService.buildSchema(SCHEMA_SEED, FORM_ID);
+            expect(actual).toMatchSnapshot();
+        });
+    });
+
+    describe("generateSchemaInstruction", () => {
+        const EXAMPLE = {
+            demande: {
+                champs: {
+                    idC3: { label: "labelC3", value: "don't care" },
+                    idC1: { label: "labelC1", value: "don't care" },
+                },
+                annotations: {
+                    idA3: { label: "labelA3", value: "don't care" },
+                    idA2: { label: "labelA2", value: "don't care" },
+                },
+            },
+        } as unknown as DemarchesSimplifieesDataEntity;
+        const USER_INPUT = "userInput";
+
+        beforeAll(() => {
+            jest.mocked(input).mockResolvedValue(USER_INPUT);
+        });
+        afterAll(() => {
+            jest.mocked(input).mockRestore();
+        });
+
+        it.each`
+            seedProperty | value
+            ${"from"}    | ${"where to get value"}
+        `("returns $seedProperty singleSeed as it is", async ({ seedProperty, value }) => {
+            const expected = { [seedProperty]: value };
+            // @ts-expect-error -- test private method
+            const actual = await demarchesSimplifieesService.generateSchemaInstruction(
```

Comment:
je comprend pas comment ça peut fonctionner : aucun lien entre "from" / "where to get value" et  les informations dans `EXAMPLE`